Lib/test/test_typing.py

```diff
@@ -3691,6 +3691,26 @@ def test_errors(self):
             class D(UserName):
                 pass
 
+    def test_or(self):
+        UserId = NewType('UserId', int)
+
+        self.assertEqual(UserId | int, Union[UserId, int])
+        self.assertEqual(int | UserId, Union[int, UserId])
+
+        self.assertEqual(get_args(UserId | int), (UserId, int))
+        self.assertEqual(get_args(int | UserId), (int, UserId))
```

Comment:
I think this doesn't test for the bug the user is reporting. In fact without this PR, these tests will all work, just instead of being ``typing.Union``, it is ``types.Union`` (because ``int`` implements ``__or__``). So the test won't be able to detect a regression.

The bug reported is that ``UserId1 | UserId2`` won't work, because they both don't implement ``__or__``.

Reply:
You are right, we should add tests to cover case when there are two `NewType`.

Reply:
I will open another PR with more tests. I was not expecting that it will be merged so fast)